### ISSUE TITLE
Ema "だけど" duplicate word bug

### ISSUE THREAD

**Issue by jzohrab** [jzohrab/lute-v3#375]
With Ema's db (sent by email), I get an error on trying to read page 7:

```
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: words.WoTextLC, words.WoLgID
[SQL: INSERT INTO words ("WoLgID", "WoText", "WoTextLC", "WoStatus", "WoTranslation", "WoRomanization", "WoTokenCount", "WoSyncStatus") VALUES (?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: (1, 'だけど', 'だけど', 0, None, None, 1, 0)]
```

Connecting directly to the db, the term already exists:

```
sqlite> .headers on
sqlite> .mode column
sqlite> select woid, wotext, wotextlc, wotokencount, wostatus from words where wotext = 'だけど';
WoID  WoText  WoTextLC  WoTokenCount  WoStatus
----  ------  --------  ------------  --------
626   だけど     だけど       1             0    
```

but reactivating some debugging print cruft in the code (blah, should have logging), it's trying to create that term again:

```
starting reading for text id = 15
creating toks ['だけど']
creating unique toks ['だけど']
all_new_terms = [<Term None 'だけど'>]
utlcs keys = dict_keys(['だけど'])
```

**Comment by jzohrab** [comment]
`ff32fdf58f0e2b80bf08476` in branch `issue_375_jp_prod_bug` dumps the text items:

```
....
<TextItem "難しい" (wo_id=410)>
<TextItem "。" (wo_id=None)>
<TextItem "だけど" (wo_id=None)>     # why no mapped word id???
<TextItem "も" (wo_id=8)>
<TextItem "、" (wo_id=None)>
<TextItem "慣れ" (wo_id=579)>
...
```

Looking into the render service.py section:

```
def get_paragraphs(s, language):
    """
    Get array of arrays of RenderableSentences for the given string s.
    """
    terms = find_all_Terms_in_string(s, language)
    print("FOUND TERMS: ----------------")
    for t in terms:
        print(t, flush=True)
    print("END FOUND TERMS -------------")
```

The search finds some terms:

```
だけど

<Term 329 'だ'>
<Term 620 'けど'>
<Term 416 'けども'>
```

But it doesn't find the だけど "dakedo" term, even though it exists in the db:

```
sqlite> select woid, wotext, wotokencount, wostatus from words where woid in (329, 626, 416, 620);
WoID  WoText  WoTokenCount  WoStatus
----  ------  ------------  --------
329   だ       1             98      
416   けども     1             0       
620   けど      1             0       
626   だけど     1             0       
```

**Comment by jzohrab** [comment]
The parser is giving inconsistent results.

`get_paragraphs` calls `find_all_Terms_in_string`.  `find_all_Terms_in_string` has the following:

```
    cleaned = re.sub(r"\s+", " ", s)
    tokens = language.get_parsed_tokens(cleaned)
    print("ALL TOKENS: --------------------")
    for t in tokens:
        print(t)
    print("END ALL TOKENS: --------------------")
```

and that gives

```
...
<"難しい" (word: True, eos: False)>
<"。" (word: False, eos: True)>
<"だ" (word: True, eos: False)>
<"けども" (word: True, eos: False)>
<"、" (word: False, eos: False)>
<"慣れ" (word: True, eos: False)>
...
```

BUT then `get_paragraphs` has the following:

```
    ParsedToken.reset_counters()
    tokens = language.get_parsed_tokens(s)
    print("IN GET_PARAGRAPHS AGAIN, ALL TOKENS: --------------------")
    for t in tokens:
        print(t)
    print("END IN GET_PARAGRAPHS AGAIN, ALL TOKENS: --------------------")
```

which gives

```
<"難しい" (word: True, eos: False)>
<"。" (word: False, eos: True)>
<"¶" (word: False, eos: True)>
<"¶" (word: False, eos: True)>
<"だけど" (word: True, eos: False)>
<"も" (word: True, eos: False)>
<"、" (word: False, eos: False)>
```

Note the parsing is NOT THE SAME the second time around, which is giving the error -- the term "だけど" is returned when getting the tokens to render on the page, but it's not getting returned when trying to find database matches.

The only difference between the two calls is the line `cleaned = re.sub(r"\s+", " ", s)`, unless mecab is somehow keeping track of the parsing calls between invocations and adjusting its output.

**Comment by jzohrab** [comment]
Fixed in develop, launching as 3.3.1.

**Comment by jzohrab** [comment]
Verified, and faster too.
